### PR TITLE
Corrected condition to auto add additional hosts

### PR DESCRIPTION
**Pull request by gobindadas**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1746395

@tiraboschi @irosenzw Please review and merge.

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: hooks/after_setup/add_host_storage_domain.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@
   with_items: "{{ gluster.hosts }}"
   loop_control:
     loop_var: host
-  when: gluster is defined and hosts in gluster and gluster.hosts | length > 1
+  when: gluster is defined and "hosts in gluster" and gluster.hosts | length > 1
```
Comment (tiraboschi) [inline hooks/after_setup/add_host_storage_domain.yml]: Please compare it with line 19

Comment (gobindadas) [reply hooks/after_setup/add_host_storage_domain.yml]: Done

**Comment by irosenzw** [comment]
@gobindadas did you verified it ?

**Comment by tiraboschi** [comment]
```shell
[stirabos@t470s pr242]$ cat test.yaml 
---
- hosts: localhost
  connection: local
  tasks:
  - name: Include Host vars
    include_vars: "vars.yaml"
  - debug: var=gluster
  - debug:
      msg: "{{ 'Processing: ' + item }}"
    with_items: "{{ gluster.hosts }}"
    when: "gluster is defined and 'hosts' in gluster and gluster.hosts | length > 1"

[stirabos@t470s pr242]$ cat vars.yaml 
gluster:
 hosts:
  rhsqa-grafton8-nic2.lab.eng.blr.redhat.com:
  rhsqa-grafton9-nic2.lab.eng.blr.redhat.com:
 vars:
  storage_domains: [{"name":"data","host":"rhsqa-grafton7.lab.eng.blr.redhat.com","address":"rhsqa-grafton7.lab.eng.blr.redhat.com","path":"/data","mount_options":"backup-volfile-servers=rhsqa-grafton8.lab.eng.blr.redhat.com:rhsqa-grafton9.lab.eng.blr.redhat.com"},{"name":"vmstore","host":"rhsqa-grafton7.lab.eng.blr.redhat.com","address":"rhsqa-grafton7.lab.eng.blr.redhat.com","path":"/vmstore","mount_options":"backup-volfile-servers=rhsqa-grafton8.lab.eng.blr.redhat.com:rhsqa-grafton9.lab.eng.blr.redhat.com"}]
[stirabos@t470s pr242]$ ansible-playbook test.yaml -i localhost,

PLAY [localhost] *************************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *******************************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Include Host vars] *****************************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [debug] *****************************************************************************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "gluster": {
        "hosts": {
            "rhsqa-grafton8-nic2.lab.eng.blr.redhat.com": null,
            "rhsqa-grafton9-nic2.lab.eng.blr.redhat.com": null
        },
        "vars": {
            "storage_domains": [
                {
                    "address": "rhsqa-grafton7.lab.eng.blr.redhat.com",
                    "host": "rhsqa-grafton7.lab.eng.blr.redhat.com",
                    "mount_options": "backup-volfile-servers=rhsqa-grafton8.lab.eng.blr.redhat.com:rhsqa-grafton9.lab.eng.blr.redhat.com",
                    "name": "data",
                    "path": "/data"
                },
                {
                    "address": "rhsqa-grafton7.lab.eng.blr.redhat.com",
                    "host": "rhsqa-grafton7.lab.eng.blr.redhat.com",
                    "mount_options": "backup-volfile-servers=rhsqa-grafton8.lab.eng.blr.redhat.com:rhsqa-grafton9.lab.eng.blr.redhat.com",
                    "name": "vmstore",
                    "path": "/vmstore"
                }
            ]
        }
    }
}

TASK [debug] *****************************************************************************************************************************************************************************************************************************************************************************
ok: [localhost] => (item=rhsqa-grafton8-nic2.lab.eng.blr.redhat.com) => {
    "msg": "Processing: rhsqa-grafton8-nic2.lab.eng.blr.redhat.com"
}
ok: [localhost] => (item=rhsqa-grafton9-nic2.lab.eng.blr.redhat.com) => {
    "msg": "Processing: rhsqa-grafton9-nic2.lab.eng.blr.redhat.com"
}

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=4    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

```

**Comment by tiraboschi** [comment]
@gobindadas can you please squash all the commits and rebase?

**Pull request merged**